api/v1alpha3/azurecluster_conversion.go

```diff
@@ -221,51 +221,52 @@ func Convert_v1beta1_NetworkSpec_To_v1alpha3_NetworkSpec(in *infrav1beta1.Networ
 }
 
 // Convert_v1beta1_VnetSpec_To_v1alpha3_VnetSpec.
-func Convert_v1beta1_VnetSpec_To_v1alpha3_VnetSpec(in *infrav1beta1.VnetSpec, out *VnetSpec, s apiconversion.Scope) error { //nolint
+func Convert_v1beta1_VnetSpec_To_v1alpha3_VnetSpec(in *infrav1.VnetSpec, out *VnetSpec, s apiconversion.Scope) error {
 	if err := autoConvert_v1beta1_VnetSpec_To_v1alpha3_VnetSpec(in, out, s); err != nil {
 		return err
 	}
 
 	// copy VnetClassSpec fields
-	out.CIDRBlocks = *(&in.CIDRBlocks)
-	out.Tags = Tags(*&in.Tags)
+	out.CIDRBlocks = in.CIDRBlocks
+	out.Tags = Tags(in.Tags)
 
 	return nil
 }
 
 // Convert_v1alpha3_SubnetSpec_To_v1beta1_SubnetSpec.
-func Convert_v1alpha3_SubnetSpec_To_v1beta1_SubnetSpec(in *SubnetSpec, out *infrav1beta1.SubnetSpec, s apiconversion.Scope) error { //nolint
+func Convert_v1alpha3_SubnetSpec_To_v1beta1_SubnetSpec(in *SubnetSpec, out *infrav1.SubnetSpec, s apiconversion.Scope) error {
 	if err := autoConvert_v1alpha3_SubnetSpec_To_v1beta1_SubnetSpec(in, out, s); err != nil {
 		return err
 	}
 
 	// Convert SubnetClassSpec fields
-	out.Role = infrav1beta1.SubnetRole(in.Role)
-	out.CIDRBlocks = *(&in.CIDRBlocks)
```

Comment:
There were several of these no-op warnings:

```go
api/v1alpha3/azurecluster_conversion.go:244:19: SA4001: *&x will be simplified to x. It will not copy x. (staticcheck)
	out.CIDRBlocks = *(&in.CIDRBlocks)
	                 ^
```

I hope the original intent wasn't to copy the field this way...